languages/elixir/concepts/regular-expressions/about.md

```diff
@@ -1 +1,103 @@
-TODO: add information on regular-expressions concept
+Regular expressions (regex) are a powerful tool for working with strings in Elixir. Regular expressions in Elixir follow the **PCRE** specification (**P**erl **C**ompatible **R**egular **E**xpressions). String patterns representing the regular expression's meaning are first compiled then used for matching all or part of a string.
+
+In Elixir, the most common way to create regular expressions is using [the `~r` sigil][sigils-regex]. Sigils provide _syntactic sugar_ shortcuts for common tasks in Elixir. In this case, `~r` is a shortcut for using `Regex.compile!/2`.
+
+```elixir
+Regex.compile!("test") == ~r/test/
+# => true
+```
+
+The `=~/2` operator is useful to perform a regex match on a string to return a `boolean` result.
+
+```elixir
+"this is a test" =~ ~r/test/
+# => true
+```
+
+## Regex syntax review
+
+- Some characters in a regular expression pattern have special meaning, to use the character plainly it must be escaped with `\`, e.g. `~r/\?/`.
+- _Character classes_ (e.g. `\d`, `\w`) allow patterns to match a range of characters
+- _Alternations_ (`|`) allow patterns to match one pattern or another
+- _Quantifiers_ (`{N, M}`, `*`, `?`) allow patterns to match a specified number of repeating patterns
+- _Groups_ (`()`) allow parts of patterns to function as a unit
+
+## Captures
```

Comment:
I rewrote the regular expression's `after.md` a lot, e.g. I added sections about captures and modifiers. It was initially written in a style that assumed it will be read immediately after solving the exercise, thus avoiding repeating some details. But now that it can be read anytime, I feel like it should be more complete and detailed.

Reply:
That makes sense!